ci/run_clang_tidy.sh

```diff
@@ -26,7 +26,7 @@ echo "Generating compilation database..."
 
 # bazel build need to be run to setup virtual includes, generating files which are consumed
 # by clang-tidy
-"${ENVOY_SRCDIR}/tools/gen_compilation_database.py" --include_headers
+"${ENVOY_SRCDIR}/tools/gen_compilation_database.py" --include_headers ${COMP_DB_TARGETS:+$COMP_DB_TARGETS}
```

Comment:
why not just `${COMP_DB_TARGETS}` ?

Reply:
see above comment about separate args

Reply:
shellcheck complains
![image](https://user-images.githubusercontent.com/63568820/112051584-7dc5cd80-8b28-11eb-8648-a3d8815d8c1b.png)

but then quoting it will register as an empty-string target in the Python script.

Reply:
So I suppose the "expand only if it is defined" is the most appropriate thing to do in this case. I think? :-)

Reply:
i think you need to what i suggested above with `read -ra`

Reply:
Legitimate question, what is the difference between that vs what I have currently?

Reply:
a) works b) shellcheck is happy

Reply:
Sure, but that's also the case with what I currently have (using `:+` syntax). Happy to make the update if it's preferred for stylistic reasons, just trying to understand the rationale for future changes.

Reply:
its the right way to do it

stepping through

- in shell you set a string
- in bash script split string into array
- in python receive args as separate args